phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexScrutinyToolPerformanceIT.java

```diff
@@ -0,0 +1,219 @@
+//                "CREATE VIEW %s AS SELECT * FROM %s", "CREATE INDEX %s ON %s (NAME) " });
+        return list;
+    }
+
+    public IndexScrutinyToolPerformanceIT(String dataTableDdl, String viewDdl, String indexTableDdl) {
+        this.dataTableDdl = dataTableDdl;
+        this.viewDdl = viewDdl;
+        this.indexTableDdl = indexTableDdl;
+    }
+
+    /**
+     * Create the test data and index tables
+     */
+    @Before public void setup() throws SQLException {
+        generateUniqueTableNames();
+        createTestTable(getUrl(), String.format(dataTableDdl, dataTableFullName));
+        if (!Strings.isNullOrEmpty(viewDdl)) {
+            createTestTable(getUrl(), String.format(viewDdl, viewFullName, dataTableFullName));
+            createTestTable(getUrl(), String.format(indexTableDdl, indexTableName, viewFullName));
+        } else {
+            createTestTable(getUrl(), String.format(indexTableDdl, indexTableName, dataTableFullName));
+        }
+        props = PropertiesUtil.deepCopy(TEST_PROPERTIES);
+        conn = DriverManager.getConnection(getUrl(), props);
+        String dataTableUpsert = String.format(UPSERT_SQL, dataTableFullName);
+        dataTableUpsertStmt = conn.prepareStatement(dataTableUpsert);
+        String viewUpsert = String.format(UPSERT_SQL, viewFullName);
+        viewUpsertStmt = conn.prepareStatement(viewUpsert);
+        conn.setAutoCommit(false);
+        testTime = EnvironmentEdgeManager.currentTimeMillis() - 1000;
+
+    }
+
+    @After public void teardown() throws SQLException {
+        if (conn != null) {
+            conn.close();
+        }
+    }
+
+    @Test
+    public void testSkipScan() throws Exception {
+        String baseTableName = dataTableName;
+        PreparedStatement upsertStmt = dataTableUpsertStmt;
+        if (!Strings.isNullOrEmpty(viewDdl)) {
+            baseTableName = viewName;
+            upsertStmt = viewUpsertStmt;
+        }
+
+        for (int i=0; i < 2; i++) {
+            if (dataTableDdl.contains("MULTI_TENANT")) {
+                upsertRowMultiTenant(upsertStmt, "val"+i, i,"name-" + i, 19999);
+            } else {
+                upsertRow(upsertStmt, i, "name-" + i, 19999);
+            }
+            conn.commit();
+        }
+
+        IndexScrutinyToolIT.runScrutiny(schemaName, baseTableName,
+                indexTableName, null, SourceTable.INDEX_TABLE_SOURCE);
+        PreparedStatement targetStmt = IndexScrutinyTool.getTargetTableQueryForScrutinyTool();
+        assertSkipScanFilter(targetStmt);
+
+        IndexScrutinyToolIT.runScrutiny(schemaName, baseTableName,
+                indexTableName, null, SourceTable.DATA_TABLE_SOURCE);
+        targetStmt = IndexScrutinyTool.getTargetTableQueryForScrutinyTool();
+        assertSkipScanFilter(targetStmt);
+    }
+
+    private void assertSkipScanFilter(PreparedStatement targetStmt) throws SQLException {
```

Comment:
Shall we have a test for measuring performance as well?

Reply:
Not sure how we can do that